owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/ReadRemoteFolderOperation.kt

```diff
@@ -0,0 +1,102 @@
+import com.owncloud.android.lib.common.http.methods.webdav.DavConstants
+import com.owncloud.android.lib.common.http.methods.webdav.DavUtils
+import com.owncloud.android.lib.common.http.methods.webdav.PropfindMethod
+import com.owncloud.android.lib.common.network.WebdavUtils
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.operations.RemoteOperationResult.ResultCode
+import timber.log.Timber
+import java.net.URL
+import java.util.ArrayList
+
+/**
+ * Remote operation performing the read of remote file or folder in the ownCloud server.
+ *
+ * @author David A. Velasco
+ * @author masensio
+ * @author David González Verdugo
+ */
+class ReadRemoteFolderOperation(
+    val remotePath: String
+) : RemoteOperation<ArrayList<RemoteFile>>() {
+
+    /**
+     * Performs the read operation.
+     *
+     * @param client Client object to communicate with the remote ownCloud server.
+     */
+    override fun run(client: OwnCloudClient): RemoteOperationResult<ArrayList<RemoteFile>> {
+        try {
+            val propfindMethod = PropfindMethod(
+                URL(client.userFilesWebDavUri.toString() + WebdavUtils.encodePath(remotePath)),
+                DavConstants.DEPTH_1,
+                DavUtils.getAllPropset()
+            )
+            client.setFollowRedirects(true)
+
+            val status = client.executeHttpMethod(propfindMethod)
+
+            if (isSuccess(status)) {
+                val mFolderAndFiles = ArrayList<RemoteFile>()
+
+                // parse data from remote folder
+                val remoteFolder = RemoteFile(propfindMethod.root, AccountUtils.getUserId(mAccount, mContext)).apply {
+                    owner = mAccount.name
+                }
+                mFolderAndFiles.add(remoteFolder)
+
+                // loop to update every child
+                propfindMethod.members.forEach { resource ->
+                    val file = RemoteFile(resource, AccountUtils.getUserId(mAccount, mContext))
+                    file.owner = mAccount.name
+                    mFolderAndFiles.add(file)
+                }
+
+                // Result of the operation
+                return RemoteOperationResult<ArrayList<RemoteFile>>(ResultCode.OK).apply {
+                    data = mFolderAndFiles
+                    Timber.i("Synchronized $remotePath with ${mFolderAndFiles.size} files. ${this.logMessage}")
+                }
+            } else { // synchronization failed
+                return RemoteOperationResult<ArrayList<RemoteFile>>(propfindMethod).also {
+                    Timber.w("Synchronized $remotePath ${it.logMessage}")
+                }
+            }
+        } catch (e: Exception) {
+            return RemoteOperationResult<ArrayList<RemoteFile>>(e).also {
+                Timber.e(it.exception, "Synchronized $remotePath")
+            }
+        }
+    }
```

Comment:
```suggestion
override fun run(client: OwnCloudClient): RemoteOperationResult<ArrayList<RemoteFile>> =
        try {
            val propfindMethod = PropfindMethod(
                URL(client.userFilesWebDavUri.toString() + WebdavUtils.encodePath(remotePath)),
                DavConstants.DEPTH_1,
                DavUtils.getAllPropset()
            )
            client.setFollowRedirects(true)

            val status = client.executeHttpMethod(propfindMethod)

            if (isSuccess(status)) {
                // Result of the operation
                RemoteOperationResult<ArrayList<RemoteFile>>(ResultCode.OK).apply {
                    val remoteFolder = RemoteFile(propfindMethod.root, AccountUtils.getUserId(mAccount, mContext)).apply {
                        owner = mAccount.name
                    }

                    data.add(remoteFolder)

                    propfindMethod.members.forEach { resource ->
                        val file = RemoteFile(resource, AccountUtils.getUserId(mAccount, mContext))
                        file.owner = mAccount.name
                        data.add(file)
                    }
                }.also {
                    Timber.i("Synchronized $remotePath with ${it.data.size} files. ${it.logMessage}")
                }

            } else { // synchronization failed
                RemoteOperationResult<ArrayList<RemoteFile>>(propfindMethod).also {
                    Timber.w("Synchronized $remotePath ${it.logMessage}")
                }
            }
        } catch (e: Exception) {
            RemoteOperationResult<ArrayList<RemoteFile>>(e).also {
                Timber.e(it.exception, "Synchronized $remotePath")
            }
        }
```

Reply:
I understand your point of view and it is less verbose, but i think we lose readability using it that way.